developers/weaviate/current/configuration/authentication.md

```diff
@@ -134,17 +123,55 @@ services:
       AUTHENTICATION_OIDC_SKIP_CLIENT_ID_CHECK: 'false'
 ```
 
+#### Weaviate OpenID endpoint
+
+If you have authentication enabled, you can request all details needed from the following endpoint:
+
+```bash
+$ curl http://localhost:8080/v1/.well-known/openid-configuration
+```
+
 ### How to use
 
-1. Obtain a valid token from the token issuer you configured.
-2. Send this token along any REST request in the Header like so: `Authorization: Bearer <token>`. Make sure to replace `<token>` with your actual token.
+OIDC authentication requires obtaining a valid token from the token issuer so that it can be sent in the header of any REST request.
+
+The OIDC standard allows for many different methods *(flows)* of obtaining a token. The appropriate method can vary depending on your situation, including configurations at the token issuer, and your requirements. 
+
+While it is outside the scope of our documentation to cover every OIDC authentication flow, some possible options are to:
+- Use the `resource owner password flow` for trusted applications. 
+- Or use the `hybrid flow` if Microsoft Azure AD is your token issuer or if you would like to prevent exposing passwords.
+
+We outline the steps below for both methods of obtaining a token.
+
+#### Resource owner password flow
+1. Send a GET request to `[WEAVIATE_URL]/v1/.well-known/openid-configuration` to fetch Weaviate's OIDC configuration (`wv_oidc_config`)
+2. Parse the `clientId` and `href` from `wv_oidc_config`
+3. Send a GET request to `href` to fetch the token issuer's OIDC configuration (`token_oidc_config`)
+4. Ensure that `password` is in the `grant_types_supported` list in `token_oidc_config`.
+  - If this is not the case, the token issuer is likely not configured for `resource owner password flow`. You may need to reconfigure the token issuer or use another method.
+5. Send a POST request to the `token_endpoint` of `token_oidc_config` with the body: 
+  - `{"grant_type": "password", "client_id": client_id, "username": [USERNAME], "password": [PASSWORD]}`. 
+  - Where `[USERNAME]` and `[PASSWORD]` are replaced with the actual values for each.
+6. Parse the response (`token_resp`), and look for `access_token` in `token_resp`. This is your Bearer token.
+
+#### Hybrid flow
+1. Send a GET request to `[WEAVIATE_URL]/v1/.well-known/openid-configuration` to fetch Weaviate's OIDC configuration (`wv_oidc_config`)
+2. Parse the `clientId` and `href` from `wv_oidc_config`
+3. Send a GET request to `href` to fetch the token issuer's OIDC configuration (`token_oidc_config`)
+4. Construct a URL (`auth_url`) with the following parameters, based on `authorization_endpoint` from `token_oidc_config`. This will look like the following:
+- `{authorization_endpoint}`?client_id=`{clientId}`&response_type=code%20id_token&response_mode=fragment&redirect_url=`{redirect_url}`&scope=openid&nonce=abcd
+- The `redirect_url` here can be any address. 
```

Comment:
I think the adress needs to be whitelisted in azure

Reply:
Looks like it is mandatory for every provider (https://openid.net/specs/openid-connect-core-1_0.html) - I have edited accordingly.

Reply:
👍